ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -436,8 +427,21 @@ private static int deleteInvalidMetadataFile(
     return count;
   }
 
-  private static String formatTime(long timestamp) {
-    return LocalDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneId.systemDefault())
-        .toString();
+
+  private static void scanManifestFiles(
+      InputFile manifestList, Consumer<ManifestFile> fileConsumer) {
+    try (CloseableIterable<ManifestFile> files =
+        Avro.read(manifestList)
```

Comment:
I'm concerned that this method of reading the manifest file relies too heavily on the structure of iceberg metadata.

Reply:
fixed.